### PR TITLE
Obs rubrics pk

### PR DESCRIPTION
**Pull request by pradeepkumarcm96**
Story - https://project-sunbird.atlassian.net/browse/SB-27420

**Newly Create Compoents**

**AssignPageNumberComponent**  - For Assigning page number of the questions this page is created to display the question list ( reference sample ui - https://www.figma.com/proto/MgE9UW1rrC4cYfcbO9RO7t/Vidyadaan?page-id=2325%3A44&node-id=2325%3A45&viewport=241%2C48%2C0.1&scaling=min-zoom&starting-point-node-id=2325%3A45)
**PlainTreeComponent** - In Assign page number component we need to show the criteria and ecm based on the observation recorded by type(self or external) on the left side we have used this component(functionalities not implemented now);

**Functionality Change** 

**EditorComponent** - new function call **fetchFrameWorkDetails** is added  to get the categoris from the framework and append to the data it will be the sent to fancy tree to render the category(i,e domain and criteria).
**FancyTreeComponent** - new condition for obs with rubrics is added and new functions such as **buildTreeFromFramework** added to construct the tree based on the framework data , **removeIntermediateLevelsFromFramework** added to remove the intermediate levels in the framework using the depth value.
**MetaFormComponent** - changes in **valueChanges** function to construct the outcomeDecleration from the levels feilds,**createLeavels** added to create the outcomeDeclaration.
**HelperService** - created **addDepthToHierarchy** to add the depth value to children.
**EditorService** - created **fetchOutComeDeclaration** to fetch the outcomeDeclaration from the questionset api.

### CODE REVIEW

File: projects/collection-editor-library/src/lib/components/assign-page-number/assign-page-number.component.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          <div class="three wide column p-0 overflow-y  mb-40 h-100vh">
+            <lib-plain-tree class="sb-collectionTree-fancyTree sb-collectionTree-fancyTreelist sb-questionset-scroll" 
+            [treeData]="treeData" (treeEmitter)="treeEventListener($event)">
+            </lib-plain-tree>
+          </div>
+          <!-- Right Player Sections -->
+          <div class="eight wide column bg-white pl-40 mb-10 w-73">
+            <table *ngIf="questions" aria-describedby="questionsTable">
+              <tr>
+                <th>Questions</th>
```
Comment (pallakartheekreddy) [inline projects/collection-editor-library/src/lib/components/assign-page-number/assign-page-number.component.html:17]: Use `label.config.json` to store labels instead of hardcoding

File: projects/collection-editor-library/src/lib/components/editor/editor.component.ts [hunk 2]
```diff
@@ -195,13 +198,22 @@ export class EditorComponent implements OnInit, OnDestroy, AfterViewInit {
     this.helperService.initialize(channel);
   }
 
-  getFrameworkDetails(categoryDefinitionData) {
+  async getFrameworkDetails(categoryDefinitionData) {
```
Comment (pallakartheekreddy) [inline projects/collection-editor-library/src/lib/components/editor/editor.component.ts:201]: Why you have added `async` this function is calling multiple place in this will it not effect?

File: projects/collection-editor-library/src/lib/components/fancy-tree/fancy-tree.component.ts [hunk 3]
```diff
@@ -298,6 +361,10 @@ export class FancyTreeComponent implements OnInit, AfterViewInit, OnDestroy {
         this.visibility.addQuestionFromLibrary = ((node.folder === true) && !_.isEmpty(_.get(hierarchylevelData, 'children')) && _.get(this.config, 'enableAddFromLibrary') === true) ? true : false;
       }
     }
+    if (_.get(this.editorService, 'editorConfig.config.renderTaxonomy') === true){
+      this.visibility.addChild=false;
```
Comment (pallakartheekreddy) [inline projects/collection-editor-library/src/lib/components/fancy-tree/fancy-tree.component.ts:365]: check indentation issue

File: projects/collection-editor-library/src/lib/components/assign-page-number/assign-page-number.component.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  selector: 'lib-assign-page-number',
+  templateUrl: './assign-page-number.component.html',
+  styleUrls: ['./assign-page-number.component.scss']
+})
+export class AssignPageNumberComponent implements OnInit {
+
+  toolbarConfig: any = {};
+  pageId = 'pagination';
+  treeData: any;
+  questions: any;
```
Comment (pallakartheekreddy) [inline projects/collection-editor-library/src/lib/components/assign-page-number/assign-page-number.component.ts:16]: variable, not used? why do we need this?

File: projects/collection-editor-library/src/lib/components/meta-form/meta-form.component.ts [hunk 5]
```diff
@@ -334,6 +318,35 @@ export class MetaFormComponent implements OnChanges, OnDestroy {
...
+    if (_.has(event, 'shuffle')) {
+      this.showShuffleMessage(event);
+    }
+    const data = _.omit(event, ['allowECM', 'levels', 'setPeriod']);
+    if (!_.isEmpty(event?.levels)) {
+      data.outcomeDeclaration = {
+        levels: this.createLeavels(event.levels)
+      };
+    }
+    event?.instance ? data.instances = { label : event?.instances } : '';
```
Comment (pallakartheekreddy) [inline projects/collection-editor-library/src/lib/components/meta-form/meta-form.component.ts:331]: why are we assigning empty string to `data.instances` ?

File: projects/collection-editor-library/src/lib/components/options/options.component.ts [hunk 6]
```diff
@@ -139,7 +139,7 @@ export class OptionsComponent implements OnInit {
 
   setScore(value, scoreIndex) {
     const obj = {
-      respone: scoreIndex,
+      response: scoreIndex,
```
Comment (pallakartheekreddy) [inline projects/collection-editor-library/src/lib/components/options/options.component.ts:142]: did we handle this in all places?

File: projects/collection-editor-library/src/lib/components/plain-tree/plain-tree.component.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import * as _ from 'lodash-es';
+declare var $: any;
+
+
+@Component({
+  selector: 'lib-plain-tree',
+  templateUrl: './plain-tree.component.html',
+  styleUrls: ['./plain-tree.component.scss']
+})
+export class PlainTreeComponent implements OnInit {
```
Comment (pallakartheekreddy) [inline projects/collection-editor-library/src/lib/components/plain-tree/plain-tree.component.ts:16]: What is the use of this component?

File: projects/collection-editor-library/src/lib/services/editor/editor.service.ts [hunk 8]
```diff
@@ -38,6 +38,8 @@ export class EditorService {
   selectedPrimaryCategory: any;
   leafParentIdentifier: any;
   questionIds = [];
+  treeData: any;
```
Comment (pallakartheekreddy) [inline projects/collection-editor-library/src/lib/services/editor/editor.service.ts:41]: variable not used

File: projects/collection-editor-library/src/lib/services/editor/editor.service.ts [hunk 9]
```diff
@@ -38,6 +38,8 @@ export class EditorService {
...
+  outcomeDeclaration: any;
```
Comment (pallakartheekreddy) [inline projects/collection-editor-library/src/lib/services/editor/editor.service.ts:42]: same here

File: src/app/data.ts [hunk 10]
```diff
@@ -151,7 +151,7 @@ export const questionSetEditorConfig = {
       lastName: 'Bhuva',
       orgIds: ['01309282781705830427']
     },
-    identifier: 'do_113476420248436736138', // do_11330102570702438417 , do_113449692707643392118, //do_113449692707643392118 , do_113460158539554816151
```
Comment (pallakartheekreddy) [inline src/app/data.ts:154]: please don't change the default data. undo this changes.

**Review by pallakartheekreddy — CHANGES_REQUESTED** [review]
Please check above comments

File: projects/collection-editor-library/src/lib/components/options/options.component.ts (hunk 6 above)
Comment (pradeepkumarcm96) [reply projects/collection-editor-library/src/lib/components/options/options.component.ts:142]: spelling is correct no other place we are using it.

File: projects/collection-editor-library/src/lib/services/editor/editor.service.ts (hunk 8 above)
Comment (pradeepkumarcm96) [reply projects/collection-editor-library/src/lib/services/editor/editor.service.ts:41]: removed

File: projects/collection-editor-library/src/lib/services/editor/editor.service.ts (hunk 9 above)
Comment (pradeepkumarcm96) [reply projects/collection-editor-library/src/lib/services/editor/editor.service.ts:42]: removed

File: src/app/data.ts (hunk 10 above)
Comment (pradeepkumarcm96) [reply src/app/data.ts]: changes reverted

File: projects/collection-editor-library/src/lib/components/meta-form/meta-form.component.ts (hunk 5 above)
Comment (pradeepkumarcm96) [reply projects/collection-editor-library/src/lib/components/meta-form/meta-form.component.ts:331]: if the creator entered the instance name it will be set else it will be empty string

File: projects/collection-editor-library/src/lib/components/assign-page-number/assign-page-number.component.ts (hunk 4 above)
Comment (pradeepkumarcm96) [reply projects/collection-editor-library/src/lib/components/assign-page-number/assign-page-number.component.ts:17]: we have declared for the functionalities to be used right now we have not implemented the functionalities

File: projects/collection-editor-library/src/lib/components/fancy-tree/fancy-tree.component.ts (hunk 3 above)
Comment (pradeepkumarcm96) [reply projects/collection-editor-library/src/lib/components/fancy-tree/fancy-tree.component.ts]: indentation fixed

File: projects/collection-editor-library/src/lib/components/editor/editor.component.ts (hunk 2 above)
Comment (pradeepkumarcm96) [reply projects/collection-editor-library/src/lib/components/editor/editor.component.ts:201]: async and await is removed we will find a alternative way

File: projects/collection-editor-library/src/lib/components/assign-page-number/assign-page-number.component.html (hunk 1 above)
Comment (pradeepkumarcm96) [reply projects/collection-editor-library/src/lib/components/assign-page-number/assign-page-number.component.html]: label configuration added

**Review by pallakartheekreddy — APPROVED** [review]
